iceoryx_binding_c/source/binding_c_publisher.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/internal/popo/ports/publisher_port_user.hpp"
+
+using namespace iox;
+using namespace iox::cxx;
+using namespace iox::popo;
+using namespace iox::capro;
+using namespace iox::mepoo;
+
+extern "C" {
+#include "iceoryx_binding_c/publisher.h"
+}
+
+PublisherPortData*
+Publisher_new(const char* service, const char* instance, const char* event, const uint64_t historyCapacity)
+{
+    return new PublisherPortData(
+        ServiceDescription{
+            IdString(TruncateToCapacity, service),
+            IdString(TruncateToCapacity, instance),
+            IdString(TruncateToCapacity, event),
```

Comment:
Is `IdString` really our preferred name for `cxx::string`? @FerdinandSpitzschnueffler What do you think?

Reply:
Isn't `IdString` just an alias for the specific type of `cxx::string` being used for IDs ?

Reply:
yes, it's just an aliases. We should think about using char* in our C-API

Reply:
Yeah we need to figure out something better ...